### ISSUE TITLE
Enable Support for Changing Field Types

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS/registry#230]
## 💡 Description

We need to be able to support the possibility a field is created in the registry at some point in time, and we need to change the type at a future point in time.

- https://github.com/NASA-PDS/registry/issues/215

## ⚙ Engineering Details
* Potentially add functionality to registry-mgr create new re-indexed index with for new mappings to take affect
  * CLI prompt that this is a significant update and no other data should be loaded during this operation

**Comment by jordanpadams** [comment]
📆 01/2024 status: To be deferred. Will be reviewed with SWG

**Comment by jordanpadams** [comment]
Deferral approved per https://github.com/NASA-PDS/pds-swg/issues/26

**Comment by jordanpadams** [comment]
📆 05/2024 status: In work. Start delayed 1 sprint due to delays in https://github.com/NASA-PDS/registry/issues/185. No impact on build expected

**Comment by jordanpadams** [comment]
📆 06/2024 status: Start delayed and additional 2 sprints due to delays in https://github.com/NASA-PDS/registry/issues/185. No impact on build expected. Not a critical task needed for this build.

**Comment by jordanpadams** [comment]
📆 07/2024 status: Start delayed an additional sprint due to delays in https://github.com/NASA-PDS/registry/issues/185. No impact on build expected. Not a critical task needed for this build.

**Comment by alexdunnjpl** [comment]
@jordanpadams was checking this intended to be part of the sweeper or not, in the end?

https://github.com/NASA-PDS/registry-sweepers/issues/148

**Comment by jordanpadams** [comment]
@alexdunnjpl correct. https://github.com/NASA-PDS/registry-sweepers/issues/148 basically does this. @tloubrieu-jpl were just talking about this and how it relates to LDDs missing from online, and how we can go back and clean that up in the future.

**Comment by alexdunnjpl** [comment]
@jordanpadams the in-progress fix for https://github.com/NASA-PDS/registry-sweepers/issues/148 will populate missing mappings, but cannot (as initially conceived) fix a mapping which exists but has the wrong type.

We _can_ support that in the first implementation, it will just mean doing additional work before the branch is PRd.

Let me know if you'd like me to front-load that feature, or do it later as an incremental update.

**Comment by jordanpadams** [comment]
@alexdunnjpl incremental update is fine. We will add this to up-next

**Comment by jordanpadams** [comment]
📆 10/2024: in work. on schedule

**Comment by jordanpadams** [comment]
📆 11/2024 status: Delayed multiple sprints due to differences between AOSS and vanilla OpenSearch. No impact on other tasks expected.

**Comment by jordanpadams** [comment]
📆 12/2024 status: Still delayed due to other efforts. No impact expected.

**Comment by jordanpadams** [comment]
📆 01/2024 status: Completed.
